src/tracks/ui/AffordanceHandle.cpp

```diff
@@ -69,17 +67,17 @@ UIHandle::Result AffordanceHandle::Click(const TrackPanelMouseEvent& evt, Audaci
 UIHandle::Result AffordanceHandle::Release(const TrackPanelMouseEvent& event, AudacityProject* pProject, wxWindow* pParent)
 {
     auto result = TimeShiftHandle::Release(event, pProject, pParent);
-    //Clip was not moved
-    if (!TimeShiftHandle::WasMoved())
+    if (!WasMoved())
     {
-        const auto track = TrackList::Get(*pProject).Lock<Track>(GetTrack());
-
-        auto& selectionState = SelectionState::Get(*pProject);
         auto& trackList = TrackList::Get(*pProject);
-        selectionState.SelectNone(trackList);
-        selectionState.SelectTrack(*track, true, true);
+        if(const auto track = TrackList::Get(*pProject).Lock<Track>(GetTrack()))
```

Comment:
You might eliminate a repetition of `TrackList::Get(*pProject)`

Reply:
Sure, thanks!